ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Monitor.hs

```diff
@@ -502,11 +502,10 @@ localRoots actions@PeerSelectionActions{ readLocalRootPeers
                    (KnownPeers.toSet knownPeers'))
 
         $ Decision {
-            decisionTrace = [TraceLocalRootPeersChanged localRootPeers
-                                                        localRootPeers']
-                         ++ if not hasOnlyBootstrapPeers'
-                               then [TraceLedgerStateJudgementChanged YoungEnough]
-                               else [],
+            decisionTrace = TraceLocalRootPeersChanged localRootPeers localRootPeers'
+                          : [ TraceLedgerStateJudgementChanged YoungEnough
+                            | not hasOnlyBootstrapPeers' && ledgerStateJudgement /= ledgerStateJudgement'
```

Comment:
It should be `bootstrapPeersFlag && not hasOnlyBoostrapPeers' && ...` (or something similar) so that it works in `Genesis` too.